src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -927,7 +927,7 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
                 {
                     // only allow 1 or 2 arguments to -replace
                     throw InterpreterError.NewInterpreterException(rval, typeof(RuntimeException), errorPosition,
-                        "BadReplaceArgument", ParserStrings.BadReplaceArgument, ignoreCase ? "-ireplace" : "-replace", rList.Count);
+                        "BadReplaceArgument", ParserStrings.BadReplaceArgument, ignoreCase ? "-replace/-ireplace" : "-creplace", rList.Count);
```

Comment:
Why not inspect the original script text to find out whether the caller explicitly invoked `-ireplace`? 

```csharp
    string operatorText = "-creplace";
    if(ignoreCase)
    {
        if(errorPosition.Text.IndexOf("ireplace", StringComparison.InvariantCultureIgnoreCase){
            operatorText = "-ireplace";
        }
        else
        {
            operatorText = "-replace";
        }
    }

    throw InterpreterError.NewInterpreterException(rval, typeof(RuntimeException), errorPosition,
        "BadReplaceArgument", ParserStrings.BadReplaceArgument, operatorText, rList.Count)

Reply:
@IISResetMe Good catch! It must have been late when I worked on this, I missed the parameter being available in `errorPosition.Text`. From what I've found, using this value directly works to generate the correct error message: 
```csharp
throw InterpreterError.NewInterpreterException(rval, typeof(RuntimeException), errorPosition,
    "BadReplaceArgument", ParserStrings.BadReplaceArgument, errorPosition.Text, rList.Count);
```
Here's an example of the errors: 
```
PS >"foo" -replace "a", "b", "c"
InvalidOperation: The -replace operator allows only two elements to follow it, not 3.

PS >"foo" -creplace "a", "b", "c"
InvalidOperation: The -creplace operator allows only two elements to follow it, not 3.

PS >"foo" -ireplace "a", "b", "c"
InvalidOperation: The -ireplace operator allows only two elements to follow it, not 3.
```

I'm a fan of this method, I believe it's honestly more readable than the ternary operator currently being used 😁

Reply:
Beautiful! 😍  

Was unsure whether the `errorPosition` extent was for the operator itself or the whole expression.